task-launcher/src/tasks/shared/helpers/initTrialSaving.js

```diff
@@ -17,7 +17,10 @@ export const initTrialSaving = (config) => {
       code.apply(fn, arguments);
     };
 
-  jsPsych.opts.on_finish = extend(jsPsych.opts.on_finish, () => {
+  jsPsych.opts.on_finish = extend(jsPsych.opts.on_finish, (data) => {
+    if (data.save_trial) {
+      config.firekit.writeTrial(data);
+    }
     config.firekit.finishRun();
```

Comment:
The logic here actually doesn't need any change. In afcStimulus.js, in the shared folder, in the on_finish function, we should check if the final data includes the corpus trial type, and if not, add it to the data object. The trial type will come from the current stimulus object. It looks like there is a trialType property being saved but perhaps it is being overwritten by jsPsych after our code.